core/frontend/src/test/render/FrameStats.test.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { expect } from "chai";
+import { Point3d, Vector3d } from "@bentley/geometry-core";
+import { IModelApp } from "../../IModelApp";
+import { IModelConnection } from "../../IModelConnection";
+import { createBlankConnection } from "../createBlankConnection";
+import { ScreenViewport } from "../../Viewport";
+import { SpatialViewState } from "../../SpatialViewState";
+import { FrameStats } from "../../render/FrameStats";
+
+describe("FrameStats", () => {
+  let imodel: IModelConnection;
+
+  before(async () => {
+    await IModelApp.startup();
+    imodel = createBlankConnection("frame-stats");
+  });
+
+  after(async () => {
+    await imodel.close();
+    await IModelApp.shutdown();
+  });
+
+  function testViewport(width: number, height: number, devicePixelRatio: number | undefined, callback: (vp: ScreenViewport) => void): void {
+    const div = document.createElement("div");
+    div.style.width = `${width}px`;
+    div.style.height = `${height}px`;
+    div.style.position = "absolute";
+    div.style.top = div.style.left = "0px";
+    document.body.appendChild(div);
+
+    const view = SpatialViewState.createBlank(imodel, new Point3d(), new Vector3d(1, 1, 1));
+    view.viewFlags.acsTriad = view.viewFlags.grid = false;
+
+    const vp = ScreenViewport.create(div, view);
+    IModelApp.viewManager.addViewport(vp);
+    expect(vp.target.debugControl).not.to.be.undefined;

```

Comment:
devicePixelRatio is relevant to this test?

Reply:
Yeah - totally unnecessary.  Thanks for the catch.  Was just copy/pasted from another test and left in.
I have now removed it.

Reply:
Can you mark this resolved if it looks good?